proto/pdpb.proto

```diff
@@ -1,303 +1,321 @@
-    optional metapb.Peer peer = 1;
+    metapb.Peer peer = 1;
 }
 
 message RegionHeartbeatResponse {
-    // Notice, Pd only allows handling reported epoch >= current pd's. 
+    ResponseHeader header = 1;
+
+    // Notice, Pd only allows handling reported epoch >= current pd's.
     // Leader peer reports region status with RegionHeartbeatRequest
-    // to pd regularly, pd will determine whether this region 
+    // to pd regularly, pd will determine whether this region
     // should do ChangePeer or not.
     // E,g, max peer number is 3, region A, first only peer 1 in A.
     // 1. Pd region state -> Peers (1), ConfVer (1).
-    // 2. Leader peer 1 reports region state to pd, pd finds the 
-    // peer number is < 3, so first changes its current region 
+    // 2. Leader peer 1 reports region state to pd, pd finds the
+    // peer number is < 3, so first changes its current region
     // state -> Peers (1, 2), ConfVer (1), and returns ChangePeer Adding 2.
     // 3. Leader does ChangePeer, then reports Peers (1, 2), ConfVer (2),
     // pd updates its state -> Peers (1, 2), ConfVer (2).
     // 4. Leader may report old Peers (1), ConfVer (1) to pd before ConfChange
-    // finished, pd stills responses ChangePeer Adding 2, of course, we must 
-    // guarantee the second ChangePeer can't be applied in TiKV. 
-    optional ChangePeer change_peer           = 1;
-    // Pd can return transfer_leader to let TiKV does leader transfer itself. 
-    optional TransferLeader transfer_leader   = 2;
-}
-
-message PutClusterConfigRequest {
-    optional metapb.Cluster cluster = 1;
-}
-
-message PutClusterConfigResponse {
+    // finished, pd stills responses ChangePeer Adding 2, of course, we must
+    // guarantee the second ChangePeer can't be applied in TiKV.
+    ChangePeer change_peer = 2;
+    // Pd can return transfer_leader to let TiKV does leader transfer itself.
+    TransferLeader transfer_leader = 3;
 }
 
 message AskSplitRequest {
-    optional metapb.Region region      = 1;
+    RequestHeader header = 1;
+
+    metapb.Region region = 2;
 }
 
 message AskSplitResponse {
-    // We split the region into two, first uses the origin 
+    ResponseHeader header = 1;
+
+    // We split the region into two, first uses the origin
     // parent region id, and the second uses the new_region_id.
     // We must guarantee that the new_region_id is global unique.
-    optional uint64 new_region_id             = 1 [(gogoproto.nullable) = false];
+    uint64 new_region_id = 2;
     // The peer ids for the new split region.
-    repeated uint64 new_peer_ids              = 2;
+    repeated uint64 new_peer_ids = 3;
+}
+
+message ReportSplitRequest {
+    RequestHeader header = 1;
+
+    metapb.Region left = 2;
+    metapb.Region right = 3;
+}
+
+message ReportSplitResponse {
+    ResponseHeader header = 1;
 }
 
 message StoreStats {
-    optional uint64 store_id             = 1 [(gogoproto.nullable) = false];
+    uint64 store_id = 1;
```

Comment:
> - nullable, if false, a field is generated without a pointer (see warning below).

See more at https://godoc.org/github.com/gogo/protobuf/gogoproto#hdr-More_Canonical_Go_Structures

-----
Here is the diff.
```diff
type StoreStats struct {
 -	StoreId uint64 `protobuf:"varint,1,opt,name=store_id,json=storeId" json:"store_id"`
 +	StoreId uint64 `protobuf:"varint,1,opt,name=store_id,json=storeId,proto3" json:"store_id,omitempty"`
```

So the option nullable is no longer needed.